require-for-sri removed from CSP in Firefox 68

a2sheppy opened the pull request:
The `Content-Security-Policy` directive `require-for-sri` is removed
in Firefox 68. Also, this patch fixes an error in the structure of the
JSON that caused most of the entries in the data for this header to
not be shown in tables.

Source:
* https://bugzilla.mozilla.org/show_bug.cgi?id=1386214

mozfreddyb commented:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for has the label "Obsolete" but not the nice note/disclaimer at the top. Maybe you can fix that (or remove the page)?

Elchi3 commented:
Changing the data structure also requires changing all compat table queries on all CSP pages. Please make sure to follow up on that, @a2sheppy.

a2sheppy commented:
> [developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for) has the label "Obsolete" but not the nice note/disclaimer at the top. Maybe you can fix that (or remove the page)?

Done.

a2sheppy commented:
> Changing the data structure also requires changing all compat table queries on all CSP pages. Please make sure to follow up on that, @a2sheppy.

I don't know where these queries are being made. Where are there references directly to members of this? I'm not sure how to find them.

I have however updated a number of places where `require-sri-for` is referenced in the docs and updated them to indicate its obsolescence.

Elchi3 commented:
I mean the compat queries. Right now these are `{{Compat("http.headers.csp.require-sri-for")}}` but this PR has changed the structure, so the query would be `{{Compat("http.headers.csp.Content-Security-Policy.require-sri-for")}}`. This affects many CSP compat macro calls across the CSP docs.

a2sheppy commented:
> I mean the compat queries. Right now these are `{{Compat("http.headers.csp.require-sri-for")}}` but this PR has changed the structure, so the query would be `{{Compat("http.headers.csp.Content-Security-Policy.require-sri-for")}}`. This affects many CSP compat macro calls across the CSP docs.

Huh. I have no idea where those would be. Why would they be on pages other than the one for require-sri-for?

a2sheppy commented:
@Elchi3 	Believe it or not, I thought this would make the compatibility tables *work*, not be broken. I was fixing an error in the JSON structure, after all. :)

a2sheppy commented:
That's what I get for fixing a massive error that I didn't make myself. ;)

vinyldarkscratch commented:
@Elchi3 I don’t know if this is an obvious answer or not, but is there a way to do a search for the old identifier throughout MDN (aka an automated method to find all the articles that use the old {{=Compat}} statement)?  I would assume there’s something lurking around, and I’m happy to assist with the fixes!

a2sheppy commented:
> @Elchi3 I don’t know if this is an obvious answer or not, but is there a way to do a search for the old identifier throughout MDN (aka an automated method to find all the articles that use the old {{=Compat}} statement)? I would assume there’s something lurking around, and I’m happy to assist with the fixes!

You can search for uses of {{Compat}} but not for {{Compat}} with specific parameters. I'm opening each page in the CSP sidebar and updating those at least.

Elchi3 commented:
> That's what I get for fixing a massive error that I didn't make myself. I'm not convinced it's my responsibility to fix all the places that relied on the erroneous structure just because I fixed the error. ;)

Well, I added this structure on purpose because the directives are not necessarily only "features" of the Content-Security-Policy header, but also appear in Content-Security-Policy-Report-Only, or can be used in the <meta http-equiv=> element and maybe other places. Also the namespace "http.headers.csp.directive" was reading nice to me. The namespace you've chosen now is fine with me, too. I will not fight over this, but in my opinion it is now your responsibility to fix the macro callers.

a2sheppy commented:
> Well, I added this structure on purpose because the directives are not necessarily only "features" of the Content-Security-Policy header, but also appear in Content-Security-Policy-Report-Only, or can be used in the <meta http-equiv=> element and maybe other places. Also the namespace "http.headers.csp.directive" was reading nice to me. The namespace you've chosen now is fine with me, too. I will not fight over this, but in my opinion it is now your responsibility to fix the macro callers.

The problem is that if you go back and look, there was a break caused because a huge chunk of content was suddenly at the wrong level of hierarchy after this stuff was added; the braces were closed at the wrong level. That's what I fixed.

Elchi3 commented:
I don't understand what was broken before. All I see is that you've changed the hierarchy to be a level deeper and directives are now under http.headers.csp.Content-Security.<directive> instead of http.headers.csp.<directive>. Can you elaborate? The diff is horrible to make sense of anything.

a2sheppy commented:
OK, looking at this further, I think I see what's happened here. There's a difference in paradigm that was not obvious here. Since these are all subdirectives of Content-Security-Policy, it seemed that they should all be located inside it, and at some point in the content, someone had tried to make them be such, but only part of the stuff was, so things were just all kinds of weird in terms of structure, and there were places that were referring to things both with and without "Content-Security-Policy" in the dotted name. I chose to move everything inside to make it consistent, and it sounds like you had originally intended for them to be outside, so I moved them the opposite direction from what you would have done.

a2sheppy commented:
The odd thing is that I'm quite certain I checked and found a huge number of these compat tables were broken because of this JSON level deviation, but looking at things now, they're all fine. Either I checked them in another dimension or they got fixed in the intervening couple of weeks...
